src/internal/operators/multicast.ts

```diff
@@ -3,54 +3,83 @@ import { Subject } from '../Subject';
-export function multicast<T>(subjectFactory: (this: Observable<T>) => Subject<T>): UnaryFunction<Observable<T>, ConnectableObservable<T>>;
-export function multicast<T, O extends ObservableInput<any>>(
-  SubjectFactory: (this: Observable<T>) => Subject<T>,
-  selector: (shared: Observable<T>) => O
 ): OperatorFunction<T, ObservedValueOf<O>>;
-/* tslint:enable:max-line-length */
 
 /**
- * Returns an Observable that emits the results of invoking a specified selector on items
- * emitted by a ConnectableObservable that shares a single subscription to the underlying stream.
+ * An operator that creates a {@link ConnectableObservable}, that when connected,
+ * with the `connect` method, will use the provided subject to multicast the values
+ * from the source to all consumers.
  *
- * ![](multicast.png)
+ * @param subjectFactory A factory that will be called to create the subject. Passing a function here
+ * will cause the underlying subject to be "reset" on error, completion, or refCounted unsubscription of
+ * the source.
+ * @deprecated This will be removed in version 8. Please use the {@link connectable} creation
+ * function, which creates a connectable observable. If you were using the {@link refCount} operator
+ * on the result of the `multicast` operator, then use the {@link share} operator, which is now
+ * highly configurable. `multicast(() => new BehaviorSubject('test'))), refCount()` is equivalent to
+ * `share({ connector: () => new BehaviorSubject('test') })`.
+ */
+export function multicast<T>(subjectFactory: () => Subject<T>): UnaryFunction<Observable<T>, ConnectableObservable<T>>;
+
+/**
+ * Because this is deprecated in favor of the {@link connect} operator, and was otherwise poorly documented,
+ * rather than duplicate the effort of documenting the same behavior, please see documentation for the
+ * {@link connect} operator.
  *
- * @param {Function|Subject} subjectOrSubjectFactory - Factory function to create an intermediate subject through
- * which the source sequence's elements will be multicasted to the selector function
- * or Subject to push source elements into.
- * @param {Function} [selector] - Optional selector function that can use the multicasted source stream
- * as many times as needed, without causing multiple subscriptions to the source stream.
- * Subscribers to the given source will receive all notifications of the source from the
- * time of the subscription forward.
- * @return {Observable} An Observable that emits the results of invoking the selector
- * on the items emitted by a `ConnectableObservable` that shares a single subscription to
- * the underlying stream.
+ * @param subjectFactory A factory that creates the subject used to multicast.
+ * @param selector A setup function to setup the multicast
+ * @deprecated To be removed in version 8. Please use the new {@link connect} operator.
+ * `multicast(fn1, fn2)` is equivalent to `connect({ connector: fn1, setup: fn2 })`.
```

Comment:
nitpick/reminder: post-bikeshedding, the `setup` prop in this comment will likely need renaming. It shows up in a few other places in docs comments, too.

Reply:
Discussed offline: Moving to `connect(fn, config)`.